src/prototype_room_external.js

```diff
@@ -260,182 +238,169 @@ Room.prototype.checkBlockedPath = function() {
+        continue;
+      }
+      // Only allow pathing through owned rooms or already reserved rooms.
+      for (let routeEntry of route) {
+        let routeRoomName = routeEntry.room;
+        if (Game.rooms[routeRoomName] === undefined) {
+          continue checkRoomsLabel;
+        }
+        let routeRoom = Game.rooms[routeRoomName];
+        if (routeRoom.controller === undefined) {
+          continue checkRoomsLabel;
         }
-        this.log('Would like to spawn reserver' + Game.rooms[this.memory.reservation.base].energyCapacityAvailable + ' ' + energyThreshold);
-        if (Game.rooms[this.memory.reservation.base].controller.level > 3 && Game.rooms[this.memory.reservation.base].energyCapacityAvailable > energyThreshold) {
-          this.log('Queuing reserver ' + this.memory.reservation.base + ' ' + JSON.stringify(reserverSpawn));
-          Game.rooms[this.memory.reservation.base].memory.queue.push(reserverSpawn);
+        if (!routeRoom.controller.my && routeRoom.memory.state !== 'Reserved') {
+          continue checkRoomsLabel;
         }
       }
-    }
-    return true;
-  }
-
-  let roomName;
-  let isReserved = function(object) {
-    if (!object.reservation) {
-      return false;
-    }
-    if (object.state != 'Reserved') {
-      return false;
-    }
-    return object.reservation.base == roomName;
-  };
-
-  for (roomName of Memory.myRooms) {
-    let room = Game.rooms[roomName];
-    if (!room) {
-      return false;
-    }
-    // TODO mark as reserved earlier, but only send sourcer
-    if (room.controller.level < 4) {
-      continue;
-    }
-
-    let distance = Game.map.getRoomLinearDistance(this.name, roomName);
-    if (distance > config.external.distance) {
-      continue;
-    }
-
-    let route = Game.map.findRoute(this.name, roomName);
-    distance = route.length;
-    if (distance > config.external.distance) {
-      continue;
-    }
-
-    if (room.memory.queue && room.memory.queue.length === 0) {
-      let reservedRooms = _.filter(Memory.rooms, isReserved);
-      if (reservedRooms.length < room.controller.level - 1) {
-        this.log('Would start to spawn');
-
-        // TODO Check paths to decide for structurer
-
-        this.memory.reservation = {
-          base: roomName,
-          tick: Game.time
-        };
-        this.memory.state = 'Reserved';
-        let reserverSpawn = {
-          role: 'reserver',
-          routing: {
-            targetRoom: this.name,
-            targetId: this.controller.id
-          },
-          level: 2
+      if (room.memory.queue && room.memory.queue.length === 0 &&
+          room.energyAvailable >= room.getEnergyCapacityAvailable()) {
+        let reservedRooms = _.filter(Memory.rooms, isReservedBy(room.name));
+        // RCL: target reserved rooms
+        let numRooms = {
+          0: 0,
+          1: 0,
+          2: 0,
+          3: 0,
+          4: 1,
+          5: 2,
+          6: 3,
+          7: 6,
+          8: 9,
```

Comment:
Didn't recognize this on the last PR. I would merge this now.

But: :-) Maybe better in the config or I like usually if it is calculable.